### PR TITLE
Depoilerplated lambda

### PR DESCRIPTION
**Pull request by armanbilge**
Ok, here's one more alternative. This is much closer to what I was envisioning: if you are willing to commit to `IO` just a _wee_ bit earlier then things get a _lot_ nicer. IMHO :)

### CODE REVIEW

File: src/main/scala/com/dwolla/rabbitmq/topology/LambdaHandler.scala [hunk 1]
```diff
@@ -17,49 +13,38 @@ import fs2.INothing
...
-    Resource.eval(Random.scalaUtilRandom[F]).flatMap { implicit random =>
-      (XRay.entryPoint[F](), KmsAlg.resource[F], httpClient[F]).tupled
-    }
+  override def handler = for {
+    ep <- Resource.pure(NoopEntrypoint[IO]())
+    kms <- KmsAlg.resource[IO]
+    http <- httpClient[IO]
+    logger <- Resource.eval(Slf4jLogger.fromName[IO](lambdaName))
+  } yield { implicit env =>
+    implicit val l = logger
```
Comment (armanbilge) [inline src/main/scala/com/dwolla/rabbitmq/topology/LambdaHandler.scala:39]: One day in Scala 3 we will make the logger implicit in the for-comprehension :)

File: src/main/scala/com/dwolla/rabbitmq/topology/LambdaHandler.scala [hunk 2]
```diff
@@ -17,49 +13,38 @@ import fs2.INothing
...
-        Kleisli { implicit env: LambdaEnv[F, RabbitMQConfig] =>
-          TracedLambda(ep) { span =>
-            LambdaHandler(
-              kms.mapK(Kleisli.liftK[F, Span[F]]).withTracing,
-              NatchezMiddleware.client(http.translate(Kleisli.liftK[F, Span[F]])(Kleisli.applyK(span)))
-            ).run(span)
-          }
-        }.run
+    TracedLambda(ep) { span =>
+      Trace.ioTrace(span).flatMap { implicit trace =>
```
Comment (armanbilge) [inline src/main/scala/com/dwolla/rabbitmq/topology/LambdaHandler.scala:42]: One of my `TracedLambda` proposals was designed to deboilerplate this:
```suggestion
    TracedLambda(ep) { implicit trace =>
```

File: src/main/scala/com/dwolla/rabbitmq/topology/LambdaHandler.scala (hunk 1 above)
Comment (bpholt) [reply src/main/scala/com/dwolla/rabbitmq/topology/LambdaHandler.scala:39]: Actually this project already has better-monadic-for enabled, so we can do it here too; it's just not as pretty. 🙂 

```scala
implicit0(logger: Logger[F]) <- Resource.eval(Slf4jLogger.fromName[F](lambdaName))
```

**Comment by bpholt** [comment]
Thanks again for looking at this @armanbilge! I'm going to close this PR since we're moving to Feral in #101.

**Pull request closed without merging**